src/order/closure.lean

```diff
@@ -158,70 +170,264 @@ lemma eq_mk₃_closed (c : closure_operator α) :
+
+namespace lower_adjoint
+
+variable (α)
+
+/-- The identity function as a lower adjoint to itself. -/
 @[simps]
-def galois_connection.closure_operator {β : Type u} [preorder β]
-  {l : α → β} {u : β → α} (gc : galois_connection l u) :
+def id [preorder α] : lower_adjoint (id : α → α) :=
+{ to_fun := λ x, x,
+  gc' := galois_connection.id }
+
+variable {α}
+
+instance [preorder α] : inhabited (lower_adjoint (_root_.id : α → α)) := ⟨id α⟩
+
+section preorder
+variables [preorder α] [preorder β] {u : β → α} (l : lower_adjoint u)
+
+instance : has_coe_to_fun (lower_adjoint u) :=
+{ F := λ _, α → β, coe := to_fun }
+
+def simps.apply : α → β := l
+
+lemma gc : galois_connection l u := l.gc'
+
+@[ext] lemma ext :
+  ∀ (l₁ l₂ : lower_adjoint u), (l₁ : α → β) = (l₂ : α → β) → l₁ = l₂
+| ⟨l₁, _⟩ ⟨l₂, _⟩ h := by { congr, exact h }
+
+@[mono] lemma monotone : monotone (u ∘ l) := l.gc.monotone_u.comp l.gc.monotone_l
+
+/-- Every element is less than its closure. This property is sometimes referred to as extensivity or
+inflationarity. -/
+lemma le_closure (x : α) : x ≤ u (l x) := l.gc.le_u_l _
+
+end preorder
+
+section partial_order
+variables [partial_order α] [preorder β] {u : β → α} (l : lower_adjoint u)
+
+/-- Every lower adjoint induces a closure operator given by the composition. This is the partial
+order version of the statement that every adjunction induces a monad. -/
+@[simps]
+def closure_operator :
   closure_operator α :=
 { to_fun := λ x, u (l x),
-  monotone' := λ x y h, gc.monotone_u (gc.monotone_l h),
-  le_closure' := gc.le_u_l,
-  idempotent' := λ x, le_antisymm (gc.monotone_u (gc.l_u_le _)) (gc.le_u_l _) }
+  monotone' := l.monotone,
+  le_closure' := l.le_closure,
+  idempotent' := λ x, show (u ∘ l ∘ u) (l x) = u (l x), by rw l.gc.u_l_u_eq_u }
 
-/--
-The Galois insertion associated to a closure operator can be used to reconstruct the closure
-operator.
+lemma idempotent (x : α) : u (l (u (l x))) = u (l x) :=
+l.closure_operator.idempotent _
 
-Note that the inverse in the opposite direction does not hold in general.
--/
+lemma le_closure_iff (x y : α) : x ≤ u (l y) ↔ u (l x) ≤ u (l y) :=
+l.closure_operator.le_closure_iff _ _
+
+end partial_order
+
+section preorder
+variables [preorder α] [preorder β] {u : β → α} (l : lower_adjoint u)
+
+/-- An element `x` is closed for the closure operator `c` if it is a fixed point for it. -/
+def closed : set α := λ x, u (l x) = x
+
+lemma mem_closed_iff (x : α) : x ∈ l.closed ↔ u (l x) = x := iff.rfl
+
+lemma closure_eq_self_of_mem_closed {x : α} (h : x ∈ l.closed) : u (l x) = x := h
+
+end preorder
+
+section partial_order
+variables [partial_order α] [partial_order β] {u : β → α} (l : lower_adjoint u)
+
+lemma mem_closed_iff_closure_le (x : α) : x ∈ l.closed ↔ u (l x) ≤ x :=
+l.closure_operator.mem_closed_iff_closure_le _
+
+@[simp] lemma closure_is_closed (x : α) : u (l x) ∈ l.closed := l.idempotent x
+
+/-- The set of closed elements for `c` is exactly its range. -/
```

Comment:
(Keeping "its range" is OK in my opinion, even though it's really the range of `u \comp l`, since we'll tend to have `u = coe`.)
```suggestion
/-- The set of closed elements for `l` is exactly its range. -/
```